registry/unit.go

```diff
@@ -61,8 +62,48 @@ func (r *EtcdRegistry) getUnitByHash(hash unit.Hash) *unit.UnitFile {
 		}
 		return nil
 	}
+	return r.unitFromEtcdNode(hash, resp.Node)
+}
+
+// getAllUnitsHashMap retrieves from the Registry all Units and returns a map of hash to Unit
+func (r *EtcdRegistry) getAllUnitsHashMap() map[string]*unit.UnitFile {
+	key := r.prefixed(unitPrefix)
+	opts := &etcd.GetOptions{
+		Recursive: true,
+	}
+	resp, err := r.kAPI.Get(r.ctx(), key, opts)
+	if err != nil {
```

Comment:
you don't return `err` so this checking serves no purpose - unless we do want to return an error from this function.